types/plotly.js/index.d.ts

```diff
@@ -513,16 +513,222 @@ export interface PlotData {
 	'marker.colorbar': {}; // TODO
 	mode: 'lines' | 'markers' | 'text' | 'lines+markers' | 'text+markers' | 'text+lines' | 'text+lines+markers' | 'none';
 	hoveron: 'points' | 'fills';
-	hoverinfo: 'all' | 'name' | 'none' | 'skip' | 'text' |
-	'x' | 'x+text' | 'x+name' |
-	'x+y' | 'x+y+text' | 'x+y+name' |
-	'x+y+z' | 'x+y+z+text' | 'x+y+z+name' |
-	'y+x' | 'y+x+text' | 'y+x+name' |
-	'y+z' | 'y+z+text' | 'y+z+name' |
-	'y+x+z' | 'y+x+z+text' | 'y+x+z+name' |
-	'z+x' | 'z+x+text' | 'z+x+name' |
-	'z+y+x' | 'z+y+x+text' | 'z+y+x+name' |
-	'z+x+y' | 'z+x+y+text' | 'z+x+y+name';
+	hoverinfo: 'all' |
```

Comment:
Are those all permutations?
I think at this point we should just declare it as `string`.